spec/requests/all_controllers_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.describe 'Controllers', type: :request do
+  let(:user) { create(:user) }
+  let(:post) { create(:post, author: user) }
+
+  describe 'GET /' do
+    it 'returns http success' do
+      get '/'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      get '/'
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes the correct placeholder text' do
+      get '/'
+      expect(response.body).to include('<h1>Welcome to the Home Page!</h1>')
+    end
+  end
+
+  describe 'GET /users' do
+    it 'returns http success' do
+      get users_path
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      get users_path
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes the correct placeholder text' do
+      get users_path
+      expect(response.body).to include('<h1>User Details:</h1>')
+    end
+  end
+
+  describe 'GET /users/:id' do
+    let(:user) { create(:user) }
+
+    it 'returns http success' do
+      get user_path(user)
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      get user_path(user)
+      expect(response).to render_template(:show)
+    end
+
+    it 'includes the correct placeholder text' do
+      get user_path(user)
+      expect(response.body).to include('<h1>Here are the posts for user</h1>')
+    end
+  end
+
+  describe 'GET /users/:user_id/posts' do
+    let(:user) { create(:user) }
+
+    it 'returns http success' do
+      get user_posts_path(user)
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      get user_posts_path(user)
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes the correct placeholder text' do
+      get user_posts_path(user)
+      expect(response.body).to include('<h1>Here are the posts for user @posts</h1>')
+    end
+  end
+
+  describe 'GET /posts/:id' do
+    let(:user) { create(:user) }
+    let(:post) { create(:post, author: user) }
+
+    it 'returns http success' do
+      get user_post_path(user, post)
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      get user_post_path(user, post)
+      expect(response).to render_template(:show)
+    end
+
+    it 'includes the correct placeholder text' do
+      get user_post_path(user, post)
+      expect(response.body).to include('<h1>Specific Post</h1>')
+    end
+  end
+end
```

Comment:
- Great Job for testing the user and post controllers, I think it would be better if you make the user and post specs in different files to make your code clean and fit the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md) 👍 
<img width="302" alt="2023-07-25_16-10" src="https://github.com/ehmaddd/blog_app/assets/52010789/c048b47c-b6fa-4f21-8b53-6c452ee130a6">